unsafe/src/main/java/org/apache/spark/unsafe/types/Interval.java

```diff
@@ -43,12 +43,13 @@
    * Finally is the unit name, ends with an optional "s".
    */
   private static String unitRegex(String unit) {
-    return "(?:\\s+(-?\\d+)\\s+" + unit + "s?)?";
+    return "(?:\\s+(-?\\d+)\\s+" + unit + "s?\\s*)?";
```

Comment:
Sort-of related question -- why is this class in 'unsafe', and why not use Joda time? the JDK already has a rich set of abstractions for time. It seems like it's only used in catalyst.